cmake/BLASFinder.cmake

```diff
@@ -88,6 +88,10 @@ if ("${CMAKE_CXX_COMPILER_ID}" STREQUAL "GNU")
     set( gnu_compiler true )
 endif()
 
+if ("${CMAKE_CXX_COMPILER_ID}" STREQUAL "IntelLLVM")
+    set( intelllvm_compiler true )
+endif()
+
 if ("${CMAKE_CXX_COMPILER_ID}" STREQUAL "Intel")
```

Comment:
See comment below about `blas_libs_list`. Could we just do:
```
"${CMAKE_CXX_COMPILER_ID}" MATCHES "Intel"
```
which should match both "Intel" and "IntelLLVM"? Then we may not need any oneMKL-specific code here.

Reply:
I prefer to keep the two names different.  
It makes sense to me to explicitly separate the Intel and IntelLLVM compilers.